### PR TITLE
feat(database): add json data type

### PR DESCRIPTION
**Pull request by MrYamous**
Part of #547 to add the `json` data type to the table statements

Latest versions of MySQL accept default value for `json` but this require a version check.
I didn't want to add this complexity (at least for this pr) as this was not done either for `text` type before

Do you think we should check if value provided as `default` is a valid json ? `json_validate` seems free to use as Tempest require 8.3+

### CODE REVIEW

**Comment by brendt** [comment]
> Do you think we should check if value provided as default is a valid json ? json_validate seems free to use as Tempest require 8.3+

Yes, that makes sense!

> Latest versions of MySQL accept default value for json but this require a version check.
> I didn't want to add this complexity (at least for this pr) as this was not done either for text type before

Yeah we can ignore this for now :)

We should also make sure that JSON columns are eventually mapped to array properties. But that's outside the scope of this PR.

File: src/Tempest/Database/src/QueryStatements/CreateTableStatement.php [hunk 1]
```diff
@@ -166,6 +166,20 @@ public function boolean(
         return $this;
     }
 
+    public function json(
+        string $name,
+        bool $nullable = false,
+        ?string $default = null
```
Comment (brendt) [inline src/Tempest/Database/src/QueryStatements/CreateTableStatement.php:172]: Let's use validate_json 👍

**Review by brendt — CHANGES_REQUESTED** [review]

**Comment by MrYamous** [comment]
`json_validate` added :) 
I'm not sure about the Exception name and the new test, let me know if this is suitable

**Review by brendt — APPROVED** [review]

**Comment by brendt** [comment]
Thanks!

**Pull request merged**